src/Core/Framework/Adapter/Twig/Node/MacroOverrideNode.php

```diff
@@ -76,14 +77,18 @@ public function compile(Compiler $compiler): void
             ->write("\$blocks = [];\n\n")
             ->write('$result =  ') // Store the result of the macro, instead of directly returning it
             ->subcompile($node)
-            ->raw("\n")
-            // customization to return actual class instead of markup
-            ->write("if (SwTwigFunction::\$macroResult !== null) {\n")
-            ->write('$result = SwTwigFunction::$macroResult;' . "\n")
-            ->write('SwTwigFunction::$macroResult = null;' . "\n")
-            ->write("}\n")
-            ->write("return \$result;\n")
-            // end of customization
+            ->raw("\n");
+        // customization to return actual class instead of markup
+        if ($code && str_contains($code, 'use Shopware\Core\Framework\Adapter\Twig\SwTwigFunction;')) {
+            $compiler
+                ->write("if (Shopware\Core\Framework\Adapter\Twig\SwTwigFunction::\$macroResult !== null) {\n")
+                ->write('$result = Shopware\Core\Framework\Adapter\Twig\SwTwigFunction::$macroResult;' . "\n")
+                ->write('Shopware\Core\Framework\Adapter\Twig\SwTwigFunction::$macroResult = null;' . "\n")
+                ->write("}\n")
+                ->write("return \$result;\n");
```

Comment:
it would be better if that would always start with an `\` otherwise it could be an relative ns